ci-operator/step-registry/osd-ccs/cluster/provision/gcp/osd-ccs-cluster-provision-gcp-commands.sh

```diff
@@ -68,6 +68,47 @@ function add_iam_policy_binding()
   done
 }
 
+CLUSTER_MACHINES_CREATED=false
+function wait_for_bootstrap() {
+  if ${CLUSTER_MACHINES_CREATED}; then
+    # There are already compute instances of the cluster on GCP.
+    return 0
+  fi
+
+  if ! which gcloud; then
+    GCLOUD_TAR="google-cloud-sdk-468.0.0-linux-x86_64.tar.gz"
+    GCLOUD_URL="https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/$GCLOUD_TAR"
+    logger "INFO" "gcloud not installed, installing from $GCLOUD_URL"
+    pushd ${HOME}
+    curl -O "$GCLOUD_URL"
+    tar -xzf "$GCLOUD_TAR"
+    export PATH=${HOME}/google-cloud-sdk/bin:${PATH}
+    popd
+  fi
+
+  # login to the service project
+  service_project_id="$(jq -r -c .project_id "${GCP_CREDENTIALS_FILE}")"
+  gcloud auth activate-service-account --key-file="${GCP_CREDENTIALS_FILE}"
+  gcloud config set project "${service_project_id}"
+
+  cmd="gcloud compute instances list --filter='name~${CLUSTER_NAME}' | grep ${CLUSTER_NAME}"
+  logger "INFO" "Running Command '${cmd}'"
+  eval "${cmd}"
+  if [ $? -ne 0 ]; then
+    logger "ERROR" "Failed to find cluster machines on GCP"
+    return 1
+  fi
+  CLUSTER_MACHINES_CREATED=true
+  return 0
+}
+
+function gather_on_failure() {
+  logger "INFO" "Trying to gather debug info of the OSD cluster..."
+  ocm get "/api/clusters_mgmt/v1/clusters/${CLUSTER_ID}/logs/install" > "${ARTIFACT_DIR}/.cluster_install.log" || echo "error: Unable to pull installation log."
+  ocm get "/api/clusters_mgmt/v1/clusters/${CLUSTER_ID}/credentials" > "${ARTIFACT_DIR}/.kubeadmin.creds" || echo "error: Unable to pull credentials."
+  logger "INFO" "Please refer to '${ARTIFACT_DIR}/' directory."
```

Comment:
We should not put the credential information in the public directory `${ARTIFACT_DIR}`.  And why you want to get the kubeadmin's kubeconfig? Regularly, the customer does have permission to get the kubeadmin's kubeconfig, we always configure the IDP, and expose the 'cluster-admin' user.